site/source/tutorials/1030_routing/index.md

```diff
@@ -0,0 +1,183 @@
+In this tutorial, we will start with a basic application with no routing. We will use Dojo 2's declarative routing to configure some routes, create `outlets` from our existing widgets and use the `Link` component to create links for the application outlets.
+
+## Prerequisites
+You can [download](../assets/1030_routing-initial.zip) the demo project and run `npm install` to get started.
+
+The `@dojo/cli` command line tool should be installed globally. Refer to the [Dojo 2 local installation](../000_local_installation/) article for more information.
+
+You also need to be familiar with TypeScript as Dojo 2 uses it extensively. For more information, refer to the [TypeScript and Dojo 2](../comingsoon.html) article.
+
+{% section %}
+
+## Configuring the router
+
+{% task 'Create the routing configuration' %}
+
+All application routes needs to be to configured when creating the router instance otherwise, entering a route will not trigger a transition within the application. The `RouteConfig` a basic object consisting off `path`, `outlet`, `defaultParams`, `defaultRoute` and `children`.
+
+The routing hierarchy is generated using by adding an array `RouteConfigs` to the children property, which get recursively processed, building up the application routing paths.
+
+{% instruction 'Include the required imports `main.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:2,3 %}
+
+{% instruction 'Define the routing configuration`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:9-31 %}
+
+{% instruction 'Now, register the router in a `registry`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:33-34 %}
+
+A helper utility above, is provided by `@dojo/routing`, that can be used to create a routing instance. The util accepts the applications routing configuration, a `registry` to define a `router` injector against and returns the `router` instance.
+
+{% instruction 'Finally, set the registry on the projector and call start on the `router` instance.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:38-41 %}
+
+{% aside 'Important!' %}
+When using a `defaultRoute` in the routing configuration, then the `router` will need to be started *after* the projector is appended.
+{% endaside %}
+
+Finally to initialize the routing, the `start` needs to be called on the `router` instance.
+
+Next, we will create `outlets` to control when our widgets are displayed.
+
+{% section %}
+
+## Routing outlets
+
+{% task 'Create outlets for the routed widgets.' %}
+
+{% aside 'Reminder' %}
+The path that is associated to an outlet name is defined by the routing configuration from the first section.
+{% endaside %}
+
+An `Outlet` is a higher order component that wraps a widget, and controls whether the widget is rendered based on if the navigated route associated to the outlets name. To keep out application organized, outlets are usually stored in a separate directory called `outlets`.
```

Comment:
higher order component  ... this should probably link somewhere in case someone doesn't know the implications of higher order components.

Reply:
"based on if the navigated route associated to the outlets name" based on if the reads weirdly